### PR TITLE
Rename global root capabilities

### PR DESCRIPTION
**Pull request by qwattash**
Moved to {kernel,userspace}_*cap naming for capabilities.
I renamed the mips-specific ones by prefixing them with `mips_` as in `mips_xkseg_cap` instead of using the `kernel_*` prefix as I would not expect those to be used outside MD code.

### CODE REVIEW

**Comment by qwattash** [comment]
Jenkins will fail because I have not merged the branch with the latest dev changes.

File: sys/cheri/cheri.h [hunk 1]
```diff
@@ -115,15 +112,13 @@ extern void * __capability userspace_sealcap;
...
-/*
- * Kernel root capability
- * has all permissions
- * This spans the whole address-space
- */
-extern void * __capability cheri_kall_capability;
+/* Root kernel capability */
+extern void * __capability kernel_root_cap;
+/* Root kernel sealing capability */
+extern void * __capability kernel_root_sealcap;
```
Comment (jrtc27) [inline sys/cheri/cheri.h:121]: This duplicates the one above.

**Review by jrtc27 — CHANGES_REQUESTED** [review]

File: sys/riscv/cheri/cheri_machdep.c [hunk 2]
```diff
@@ -53,18 +53,18 @@ cheri_init_capabilities(void * __capability kroot)
 	ctemp = cheri_setaddress(kroot, CHERI_SEALCAP_KERNEL_BASE);
 	ctemp = cheri_setbounds(kroot, CHERI_SEALCAP_KERNEL_LENGTH);
 	ctemp = cheri_andperm(kroot, CHERI_SEALCAP_KERNEL_PERMS);
-	kernel_sealcap = ctemp;
+        kernel_root_sealcap = ctemp;
```
Comment (brooksdavis) [inline sys/riscv/cheri/cheri_machdep.c:56]:
```suggestion
	kernel_root_sealcap = ctemp;
```

File: sys/riscv/cheri/cheri_machdep.c [hunk 3]
```diff
@@ -74,7 +74,8 @@ cheri_init_capabilities(void * __capability kroot)
 	ctemp = cheri_andperm(ctemp, CHERI_PERMS_KERNEL_DATA);
 	devmap_init_capability(ctemp);
 
-	cheri_kall_capability = kroot;
+        kernel_root_cap = cheri_andperm(kroot,
```
Comment (brooksdavis) [inline sys/riscv/cheri/cheri_machdep.c:77]:
```suggestion
	kernel_root_cap = cheri_andperm(kroot,
```

File: sys/cheri/cheri.h (hunk 1 above)
Comment (qwattash) [reply sys/cheri/cheri.h:121]: Messed up a rename, sorry about that.

File: sys/cheri/cheri_usercap.c [hunk 4]
```diff
@@ -36,10 +36,10 @@
 #include <cheri/cheric.h>
 
 /* Set to -1 to prevent it from being zeroed with the rest of BSS */
-void * __capability userspace_cap = (void * __capability)(intcap_t)-1;
+void * __capability userspace_root_cap = (void * __capability)(intcap_t)-1;
 
 /*
- * Build a new userspace capability derived from userspace_cap.
+ * Build a new userspace capability derived from userspace_root_cap
```
Comment (brooksdavis) [inline sys/cheri/cheri_usercap.c:42]: Removing periods from these sentences seems odd.

File: sys/mips/cheri/cheri.c [hunk 5]
```diff
@@ -279,7 +285,7 @@ cheri_cpu_startup(void)
 {
 
 	/*
-	 * Documentary assertions for userspace_cap.  Default data and
+	 * Documentary assertions for userspace_root_cap  Default data and
```
Comment (brooksdavis) [inline sys/mips/cheri/cheri.c:288]:
```suggestion
	 * Documentary assertions for userspace_root_cap.  Default data and
```

File: sys/vm/vm_mmap.c [hunk 6]
```diff
@@ -455,7 +455,7 @@ kern_mmap(struct thread *td, uintptr_t addr0, size_t len, int prot, int flags,
 		.mr_pos = pos,
 #ifdef __CHERI_PURE_CAPABILITY__
 		/* Needed for fixed mappings */
-		.mr_source_cap = userspace_cap
+		.mr_source_cap = userspace_root_cap
```
Comment (brooksdavis) [inline sys/vm/vm_mmap.c:458]:
```suggestion
		.mr_source_cap = userspace_root_cap,
```
style prefers trailing commas to reduce future diffs.

**Review by brooksdavis — COMMENTED** [review]
A few small punctuation suggestions, but otherwise looks good to me.

**Pull request merged**